File: e_DSA/b_DataStructures/a_Array/SquaresOfSortedArray/Solution2.js
// 977. Squares of a Sorted Array => https://leetcode.com/problems/squares-of-a-sorted-array/description/

const sort = function (arr) {
    let newArr = []
    let j = arr.length - 1
    let i = 0

    for (let n = j; n > -1; n--) {

        if (arr[i] ** 2 > arr[j] ** 2) {
            newArr[n] = arr[i] ** 2;
            i++;
        }
        else {
            newArr[n] = arr[j] ** 2
            j--
        }

    }

    return newArr
}

let nums = [-9, -7, -5, 0, 2, 3, 5, 7]


console.log(sort(nums));

File: e_DSA/b_DataStructures/a_Array/ValidAnagram/Solution2.java
// 242. Valid Anagram => https://leetcode.com/problems/valid-anagram/description/

package e_DSA.b_DataStructures.a_Array.ValidAnagram;

import java.util.Arrays;

public class Solution2 {

    public static boolean isAnagram(String s, String t) {

        if(s.length() != t.length()) {
            return false;
        } else {
            char[] ss = s.toCharArray();
            char[] tt = t.toCharArray();
            Arrays.sort(ss);
            Arrays.sort(tt);
            
            for(int i = 0; i < ss.length; i++) {
                if(ss[i]!=tt[i])
                    return false;
            }

            return true;   
        }
    }

    public static void main(String args[]) {
        String s = "anagram", t = "nagaram";
        String r = "rat", c = "car";

        System.out.println(isAnagram(s,t));
        System.out.println(isAnagram(r, c));
        
    }

}
